dataservice/api/common/schemas.py

```diff
@@ -0,0 +1,49 @@
+from dataservice.extensions import ma
+from marshmallow import post_dump
+from flask_marshmallow import Schema
+
+
+class BaseSchema(ma.ModelSchema):
+
+    def __init__(self, code=200, message='success', *args, **kwargs):
+        self.status_code = code
+        self.status_message = message
+        super(BaseSchema, self).__init__(*args, **kwargs)
+
+    class Meta:
+        fields = ('kf_id',)
+
+    @post_dump(pass_many=True)
+    def wrap_envelope(self, data, many):
+        if type(data) is not list:
+            _links = data['_links']
+            del data['_links']
+        else:
+            _links = {}
+
+        return {'results': data,
+                '_links': _links,
+                '_status': {'message': self.status_message,
+                            'code': self.status_code}}
+
+
+class ErrorSchema(Schema):
```

Comment:
Should prolly make this the base status schema and figure out how to inherit it in the model schemas.